vllm/distributed/parallel_state.py

```diff
@@ -559,7 +563,7 @@ def init_distributed_environment(
             use_custom_allreduce=False,
         )
     else:
-        assert _WORLD.world_size == torch.distributed.get_world_size(), (
+        assert _WORLD.world_size == world_size, (
```

Comment:
Why is this change necessary? We should add a comment on the intention here as well.

Reply:
I thought about it a little more, and it would be better to completely remove the world_group patch.
It seems like these unclear issues have arisen here because the world_group patch is not absolutely necessary.

Please check the updated code that patches only tp_group and leaves world_group untouched.